packages/rum-core/src/common/context.js

```diff
@@ -0,0 +1,166 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import Url from './url'
+import { PAGE_LOAD } from './constants'
+import { getPageMetadata, getServerTimingInfo } from './utils'
+
+const LEFT_SQUARE_BRACKET = 91 // [
+const RIGHT_SQUARE_BRACKET = 93 // ]
+
+/**
+ * Both Navigation and Resource timing level 2 exposes these below information
+ *
+ * for CORS requests without Timing-Allow-Origin header, transferSize & encodedBodySize will be 0
+ */
+function getResponseContext(perfTimingEntry) {
+  const {
+    transferSize,
+    encodedBodySize,
+    decodedBodySize,
+    serverTiming
+  } = perfTimingEntry
+
+  const respContext = {
+    transfer_size: transferSize,
+    encoded_body_size: encodedBodySize,
+    decoded_body_size: decodedBodySize
+  }
+  const serverTimingStr = getServerTimingInfo(serverTiming)
+  if (serverTimingStr) {
+    respContext.headers = {
+      'server-timing': serverTimingStr
+    }
+  }
+  return respContext
+}
+
+function getDestination(parsedUrl) {
+  const { port, protocol, hostname, origin } = parsedUrl
+
+  const isDefaultSchemeAndPort =
+    (protocol === 'http:' && port === '80') ||
+    (protocol === 'https:' && port === '443')
+
+  /**
+   * If hostname begins with [ and ends with ] then its an IPV6 address
+   *
+   * since address and port are recorded separately, we are recording the
+   * info in canonical form without square brackets
+   *
+   * IPv6 check is done here instead of in the URL parser to keep the URL parser
+   * functionality minimal and it also makes things easier for service construction
+   */
+  const ipv6Hostname =
+    hostname.charCodeAt(0) === LEFT_SQUARE_BRACKET &&
+    hostname.charCodeAt(hostname.length - 1) === RIGHT_SQUARE_BRACKET
+
+  let address = hostname
+  if (ipv6Hostname) {
+    address = hostname.slice(1, -1)
+  }
+
+  return {
+    service: {
+      name: !isDefaultSchemeAndPort ? origin : protocol + '//' + hostname,
+      resource: hostname + ':' + port,
+      type: 'external'
```

Comment:
this should be 'resource' for resource-type spans

Reply:
I actually thought about this and left it to `external` since we make a HTTP request anyways for fetching the resource. wouldn't this confuse users when using service maps feature that some are resource and some are external?

Reply:
Changed to `resource` for now. I also added the comment in the master issue if its okay to keep it as `resource`

Reply:
Yeah, potentially should be the same regardless of the reason for the request. I saw your question on the main issue, so let's see what Ron thinks.